### PR TITLE
chore: Refactoring

### PR DESCRIPTION
**Pull request by tolusha**
### What does this PR do?
Refactoring

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/19284

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#updating-custom-resource-definition-file)
- [ ] [OLM bundles are up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: pkg/deploy/server/default_reconciler.go [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+	pvcClaimSize := util.GetValue(ctx.CheCluster.Spec.Storage.PvcClaimSize, deploy.DefaultPvcClaimSize)
+	if len(ctx.CheCluster.Spec.Storage.PvcClaimSize) < 1 {
+		ctx.CheCluster.Spec.Storage.PvcClaimSize = pvcClaimSize
+		if err := deploy.UpdateCheCRSpec(ctx, "pvc claim size", pvcClaimSize); err != nil {
+			return reconcile.Result{}, false, err
+		}
+	}
+
+	// This is only to correctly  manage defaults during the transition
+	// from Upstream 7.0.0 GA to the next
```
Comment (mmorhun) [inline pkg/deploy/server/default_reconciler.go:200]: Is it actual now?

File: pkg/deploy/server/default_reconciler.go [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+	"strconv"
+
+	"github.com/eclipse-che/che-operator/pkg/deploy"
+	"github.com/eclipse-che/che-operator/pkg/util"
+	"github.com/sirupsen/logrus"
+	appsv1 "k8s.io/api/apps/v1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type DefaultValuesReconciler struct {
```
Comment (mmorhun) [inline pkg/deploy/server/default_reconciler.go:25]: Name the file accordingly, please

**Review by mmorhun — APPROVED** [review]

File: pkg/deploy/consolelink/consolelink.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+func (c *ConsoleLinkReconciler) Reconcile(ctx *deploy.DeployContext) (reconcile.Result, bool, error) {
+	if !util.IsOpenShift4 || !util.HasK8SResourceObject(ctx.ClusterAPI.DiscoveryClient, ConsoleLinksResourceName) {
+		// console link is supported only on OpenShift >= 4.2
+		logrus.Debug("Console link won't be created. Consolelinks is not supported by OpenShift cluster.")
+		return reconcile.Result{}, true, nil
+	}
+
+	if !ctx.CheCluster.Spec.Server.TlsSupport {
```
[the remark this answers is not in the document]
Comment (tolusha) [reply pkg/deploy/consolelink/consolelink.go:47]: No, but it it is still in the codebase. Will be removed later.

File: pkg/deploy/identity-provider/identity_provider_reconciler.go [hunk 4]
```diff
@@ -45,26 +50,53 @@ var (
...
 		}
 	}
 
-	return true, nil
+	return reconcile.Result{}, true, nil
+}
+
+func (ip *IdentityProviderReconciler) Finalize(ctx *deploy.DeployContext) error {
+	oAuthClientName := ctx.CheCluster.Spec.Auth.OAuthClientName
+	if oAuthClientName != "" {
```
[the remark this answers is not in the document]
Comment (tolusha) [reply pkg/deploy/identity-provider/identity_provider_reconciler.go:96]: I got it

File: pkg/deploy/server/server_reconciler.go [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	AvailableStatus               = "Available"
+	UnavailableStatus             = "Unavailable"
+	RollingUpdateInProgressStatus = "Available: Rolling update in progress"
+)
+
+type ServerReconciler struct {
```
[the remark this answers is not in the document]
Comment (tolusha) [reply pkg/deploy/server/server_reconciler.go:31]: ok

File: pkg/deploy/server/default_reconciler.go (hunk 1 above)
Comment (tolusha) [reply pkg/deploy/server/default_reconciler.go:200]: I  think no, but let's keep it

File: controllers/che/checluster_controller.go [hunk 6]
```diff
@@ -103,6 +102,19 @@ func NewReconciler(
...
+	// resources since che host is used for dashboard deployment and che config map
+	reconcileManager.RegisterReconciler(server.NewCheHostReconciler())
+	reconcileManager.RegisterReconciler(postgres.NewPostgresReconciler())
+	reconcileManager.RegisterReconciler(identityprovider.NewIdentityProviderReconciler())
+	reconcileManager.RegisterReconciler(devfileregistry.NewDevfileRegistryReconciler())
+	reconcileManager.RegisterReconciler(pluginregistry.NewPluginRegistryReconciler())
+	reconcileManager.RegisterReconciler(dashboard.NewDashboardReconciler())
+	reconcileManager.RegisterReconciler(gateway.NewGatewayReconciler())
+	reconcileManager.RegisterReconciler(server.NewCheServerReconciler())
+	reconcileManager.RegisterReconciler(consolelink.NewConsoleLinkReconciler())
```
Comment (AndrienkoAleksandr) [inline controllers/che/checluster_controller.go:117]: Minor: Maybe, we can check here if Openshift 4, then register reconsiler.

File: pkg/deploy/dashboard/dashboard.go [hunk 7]
```diff
@@ -33,97 +34,98 @@ var (
...
 }
 
-func (d *Dashboard) Reconcile() (done bool, err error) {
+func (d *DashboardReconciler) Reconcile(ctx *deploy.DeployContext) (reconcile.Result, bool, error) {
 	// Create a new dashboard service
-	done, err = deploy.SyncServiceToCluster(d.deployContext, d.component, []string{"http"}, []int32{8080}, d.component)
+	done, err := deploy.SyncServiceToCluster(ctx, d.getComponentName(ctx), []string{"http"}, []int32{8080}, d.getComponentName(ctx))
 	if !done {
-		return false, err
+		return reconcile.Result{}, false, err
```
Comment (AndrienkoAleksandr) [inline pkg/deploy/dashboard/dashboard.go:53]: Do we need reconcile.Result{Requeue: true} here ?

Comment (tolusha) [reply pkg/deploy/dashboard/dashboard.go:53]: No. In general we need Requeue if we deal with non cached object (cluster ones).

File: controllers/che/checluster_controller.go (hunk 6 above)
Comment (tolusha) [reply controllers/che/checluster_controller.go:117]: Make sense

File: pkg/deploy/server/default_values_reconciler.go [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+}
+
+func NewDefaultValuesReconciler() *DefaultValuesReconciler {
+	return &DefaultValuesReconciler{}
+}
+
+func (p *DefaultValuesReconciler) Reconcile(ctx *deploy.DeployContext) (reconcile.Result, bool, error) {
+	cheFlavor := deploy.DefaultCheFlavor(ctx.CheCluster)
+	cheNamespace := ctx.CheCluster.Namespace
+	if len(ctx.CheCluster.Spec.Server.CheFlavor) < 1 {
```
Comment (AndrienkoAleksandr) [inline pkg/deploy/server/default_values_reconciler.go:36]: Up to you, but for me looks better len(ctx.CheCluster.Spec.Server.CheFlavor) == 0

File: pkg/deploy/server/server_reconciler.go [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+		logrus.Info("Found legacy custom ConfigMap. Adding those values to CheCluster.Spec.Server.CustomCheProperties")
+
+		if ctx.CheCluster.Spec.Server.CustomCheProperties == nil {
+			ctx.CheCluster.Spec.Server.CustomCheProperties = make(map[string]string)
+		}
+		for k, v := range customConfigMap.Data {
+			ctx.CheCluster.Spec.Server.CustomCheProperties[k] = v
+		}
+
+		err := ctx.ClusterAPI.Client.Update(context.TODO(), ctx.CheCluster)
```
Comment (AndrienkoAleksandr) [inline pkg/deploy/server/server_reconciler.go:121]: Maybe we can use deploy.UpdateCheCRSpec here.

**Pull request closed without merging**